### ISSUE TITLE
transformations tutorial error

### ISSUE THREAD

**Issue by jingpengwu** [JuliaGL/GLAbstraction.jl#56]
I tried to test transformations1.jl and 2.jl
`julia transformations2.jl `

there is no kitten.png and puppy.png.
```
WARNING: bytestring(p::Union{Ptr{Int8},Ptr{UInt8}}) is deprecated, use unsafe_string(p) instead.
 in depwarn(::String, ::Symbol) at ./deprecated.jl:64
 in bytestring(::Ptr{UInt8}) at ./deprecated.jl:50
 in error(::ImageMagick.MagickWand) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/libmagickwand.jl:150
 in readimage(::ImageMagick.MagickWand, ::String) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/libmagickwand.jl:231
 in #load_#14(::Type{T}, ::String, ::Bool, ::Function, ::String) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/ImageMagick.jl:80
 in #load#7(::Array{Any,1}, ::Function, ::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/ImageMagick.jl:65
 in load(::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/ImageMagick.jl:65
 in #load#17(::Array{Any,1}, ::Function, ::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:85
 in load(::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:73
 in #load#13(::Array{Any,1}, ::Function, ::String) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:45
 in load(::String) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:45
 in include_from_node1(::String) at ./loading.jl:488
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /usr/people/jingpeng/.julia/v0.5/GLAbstraction/tutorials/transformations2.jl, in expression starting on line 4
Error encountered while loading "/usr/people/jingpeng/.julia/v0.5/GLAbstraction/tutorials/images/kitten.png".
Fatal error:
ERROR: LoadError: unable to open file `/usr/people/jingpeng/.julia/v0.5/GLAbstraction/tutorials/images/kitten.png' @ error/png.c/ReadPNGImage/3667
 in error(::ImageMagick.MagickWand) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/libmagickwand.jl:152
 in readimage(::ImageMagick.MagickWand, ::String) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/libmagickwand.jl:231
 in #load_#14(::Type{T}, ::String, ::Bool, ::Function, ::String) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/ImageMagick.jl:80
 in #load#7(::Array{Any,1}, ::Function, ::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/ImageMagick.jl:65
 in load(::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/ImageMagick/src/ImageMagick.jl:65
 in #load#17(::Array{Any,1}, ::Function, ::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:85
 in load(::FileIO.File{FileIO.DataFormat{:PNG}}) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:73
 in #load#13(::Array{Any,1}, ::Function, ::String) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:45
 in load(::String) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:45
 in include_from_node1(::String) at ./loading.jl:488
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /usr/people/jingpeng/.julia/v0.5/GLAbstraction/tutorials/transformations2.jl, in expression starting on line 4
```

after copying two alternative images, I got a shader error.
```
julia transformations2.jl 
ERROR: LoadError: MethodError: Cannot `convert` an object of type FileIO.File{FileIO.DataFormat{:GLSLShader}} to an object of type GLAbstraction.Shader
This may have arisen from a call to the constructor GLAbstraction.Shader(...),
since type constructors fall back to convert methods.
 in load(::FileIO.File{FileIO.DataFormat{:GLSLShader}}) at /usr/people/jingpeng/.julia/v0.5/GLAbstraction/src/GLShader.jl:77
 in #load#13(::Array{Any,1}, ::Function, ::String) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:45
 in load(::String) at /usr/people/jingpeng/.julia/v0.5/FileIO/src/loadsave.jl:45
 in include_from_node1(::String) at ./loading.jl:488
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /usr/people/jingpeng/.julia/v0.5/GLAbstraction/tutorials/transformations2.jl, in expression starting on line 50
```
it seems that the Shader construction function for FileIO was not defined.
https://github.com/JuliaGL/GLAbstraction.jl/blob/master/src/GLShader.jl#L77

**Comment by SimonDanisch** [comment]
Sorry about that!
This commit should have fixed things: f28f7c8ef5964f3f65fce14acfc903f381188232
Please reopen if not!
I've also added a `run_all.jl`, to go through all examples. As you can see, there is an `include("downloads.jl")` at the beginning, which downloads the images. That is also mentioned in `tutorials/README.md`

**Comment by jingpengwu** [comment]
It works great! Thanks a lot! 
Also, thank you for all your impressive work! @SimonDanisch
